activesupport/test/cache/behaviors/cache_store_behavior.rb

```diff
@@ -498,7 +500,41 @@ def test_cache_miss_instrumentation
     ActiveSupport::Notifications.unsubscribe "cache_read.active_support"
   end
 
+  def test_nil_is_significant
+    skip "does not support blank key" if does_not_support_blank_keys?
+
+    @cache.write(nil, "baz")
+    assert_equal "baz", @cache.read(nil)
+
+    @cache.write([nil], "bizz")
+    assert_equal "bizz", @cache.read(nil)
+  end
+
+  def test_nil_is_significant_in_array_and_hash
+    @cache.write([ :a, nil, :b, "", :c ], "foo")
+    assert_equal "foo", @cache.read("a//b//c")
+
+    @cache.write({ hello: nil, world: "today", nil: "rules" }, "bar")
+    assert_equal "bar", @cache.read("hello=/nil=rules/world=today")
+  end
+
+  def test_empty_string_behavior
+    @cache.write(["", "", ""], "foo")
+    assert_equal "foo", @cache.read("//")
+
+    skip "does not support blank key" if does_not_support_blank_keys?
+    @cache.write([""], "bar")
+    assert_equal "bar", @cache.read("")
+
+    @cache.write("", "baz")
+    assert_equal "baz", @cache.read("")
+  end
+
   private
+    def does_not_support_blank_keys?
+      @cache.is_a?(ActiveSupport::Cache::FileStore) ||
+        @cache.is_a?(ActiveSupport::Cache::MemCacheStore)
+    end
```

Comment:
Let's leave these for the other PR and rely on Key unit tests, otherwise.

Reply:
Sounds good. I plan on deleting this commit and porting these to unit tests. Thanks for the review!

Reply:
I already had all the unit tests proposed in the other PR. I deleted some of the tests that produced a `nil` or `empty` key.

Reply:
I already had all the unit tests proposed in the other PR. I deleted some of the tests that produced a `nil` or `empty` key.

Reply:
I already had all the unit tests proposed in the other PR. I deleted some of the tests that produced a `nil` or `empty` key.